src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,259 @@
+
+        time_data = data_source.time
+
+        results_dataframe = pl.DataFrame({"time": time_data})
+
+        for field_name, field_value in self:
+            if field_value is not None:
+                if (
+                    use_epoch1_time
+                    and field_name
+                    not in ["enmo", "anglez", "sleep_windows", "nonwear_array"]
+                ) or (not use_epoch1_time and field_name in ["enmo", "anglez"]):
+                    results_dataframe = results_dataframe.with_columns(
+                        pl.Series(field_name, field_value.measurements)
+                    )
+
+        return results_dataframe
+
+
+def format_sleep_data(
+    sleep_windows: List[analytics.SleepWindow], reference_measure: models.Measurement
+) -> np.ndarray:
+    """Formats sleep windows into an array for saving.
+
+    Args:
+        sleep_windows: The list of time stamp pairs indicating periods of sleep.
+        reference_measure: The measure from which the temporal resolution will be taken.
+            Any epoch1 measure can be used, but enmo is used for consistency.
+
+    Returns:
+        1-D binary np.ndarray, with 1 indicating sleep. Will be of the same length as
+            the timestamps in the epoch1_measure.
+    """
+    sleep_array = np.zeros(len(reference_measure.time))
+
+    for window in sleep_windows:
+        sleep_mask = (reference_measure.time >= window.onset) & (
+            reference_measure.time <= window.wakeup
+        )
+        sleep_array[sleep_mask] = 1
+
+    return sleep_array
+
+
+def format_nonwear_data(
+    nonwear_data: models.Measurement, reference_measure: models.Measurement
+) -> np.ndarray:
+    """Format nonwear data into an array for saving.
+
+    Args:
+        nonwear_data: The nonwear measurement.
+        reference_measure: The measure from which the temporal resolution will be taken.
+            Any epoch1 measure can be used, but enmo is used for consistency.
+
+    Returns:
+        1-D binary np.ndarray, with 1 indicating nonwear. Will be of the same length as
+            the timestamps in the reference_measure.
+
+    """
+    nonwear_df = pl.DataFrame(
+        {
+            "nonwear": nonwear_data.measurements.astype(np.int64),
+            "time": nonwear_data.time,
+        }
+    ).set_sorted("time")
+
+    nonwear_upsample = nonwear_df.upsample(time_column="time", every="5s").fill_null(
+        strategy="forward"
+    )
+
+    end_sequence = pl.repeat(
+        nonwear_upsample["nonwear"][-1],
+        n=(len(reference_measure.time) - len(nonwear_upsample["time"])),
+        dtype=pl.Int64,
+        eager=True,
+    )
+    full_nonwear_array = pl.concat([nonwear_upsample["nonwear"], end_sequence])
+
+    return full_nonwear_array.to_numpy()
+
+
+def run(
+    input: pathlib.Path,
+    output: Optional[pathlib.Path] = None,
+    settings: config.Settings = config.Settings(),
+    calibrator: calibration.Calibration = calibration.Calibration(),
```

Comment:
I'd use a literal string here (`Literal["ggir"]`, or whatever an appropriate name is). Part of the goal of an orchestrator is to make things as easy on the user as possible; this set-up forces them to know where to import calibrators from in the first place.

One asterisk: if your intent is to allow users to provide their own calibrators, then we should discuss. In that particular case a setup very similar to this would be OK.

Reply:
One thing that is changing (potentially) is that there will now be `calibration.Calibration.method = 'GGIR'` to use the GGIR method. Otherwise all other options will go to the new default (zscore for initial guess + scipy.optimize)